src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoNullable.cs

```diff
@@ -79,6 +81,55 @@ protected override void OnWrite(ref WriteStackFrame current, Utf8JsonWriter writ
             }
         }
 
+        protected override void OnWriteDictionary(ref WriteStackFrame current, Utf8JsonWriter writer)
+        {
+            if (Converter == null)
+            {
+                return;
+            }
+
+            Debug.Assert(current.CollectionEnumerator != null);
+
+            string key;
+            TProperty? value;
+            if (current.CollectionEnumerator is IEnumerator<KeyValuePair<string, TProperty?>> enumerator)
+            {
+                // Avoid boxing for strongly-typed enumerators such as returned from IDictionary<string, TRuntimeProperty>
+                value = enumerator.Current.Value;
+                key = enumerator.Current.Key;
+            }
+            else if (current.IsIDictionaryConstructible || current.IsIDictionaryConstructibleProperty)
+            {
+                value = (TProperty?)((DictionaryEntry)current.CollectionEnumerator.Current).Value;
+                key = (string)((DictionaryEntry)current.CollectionEnumerator.Current).Key;
+            }
+            else
+            {
+                throw new NotSupportedException();
```

Comment:
I am curios about when we hit this.

Reply:
We shouldn't - removed. In the logic for non-nullable dictionary values, we hit it when we try to serialize non-generic dictionaries: https://github.com/dotnet/corefx/issues/41034.